transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem14Parser.java

```diff
@@ -110,6 +113,11 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) th
             }
             readElement(reader, element, list, subsystem, logStoreOperation);
         }
+
+        if(!relativeToEncountered) {
+            TransactionSubsystemRootResourceDefinition.OBJECT_STORE_RELATIVE_TO.parseAndSetParameter("jboss.server.data.dir", subsystem, reader);
```

Comment:
@bstansberry I have updated to add the boolean flag relativeToHasDefualtValue in the TransactionSubsystem14Parser and set it to false in the TransactionSubsystem31Parser
can you review these changes ?